lib/comma/data_extractor.rb

```diff
@@ -29,12 +29,25 @@ def yield_block_with_value(value, &block)
       end
 
       def extract_value(method)
-        extraction_object.send(method)
+        output = extraction_object.send(method)
+        time_zone_adjusted_value(output)
       end
 
       def extraction_object
         @instance
       end
+
+      def should_adjust_time_zone?(object)
+        object.respond_to?(:in_time_zone) && extraction_object.respond_to?(:site)
```

Comment:
Sure, great idea. We can even delegate `:current_time_zone` to whatever we want in the original classes/models/